src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritHandler.java

```diff
@@ -386,10 +536,28 @@ private boolean ignoreEvent(CommentAdded event) {
      * @param join if the method should wait for the thread to finish before returning.
      */
     public void shutdown(boolean join) {
-        for (EventThread worker : workers) {
-            worker.shutdown();
+        if (executor != null) {
+            ThreadPoolExecutor pool = executor;
+            executor = null;
```

Comment:
there are places throughout in the code that are assuming a non null executor. See above comments but I have most likely not found them all.

Reply:
actually executor can't be null. It is initialized in the constructor.